language/functional_tests/tests/testsuite/generics/signatures/unrestricted_as_resource.mvir

```diff
@@ -0,0 +1,10 @@
+module M {
+    struct S<T: resource> {}
+
+    foo() {
+        let x: Self.S<bool>;
+        return;
+    }
+}
+
+// check: CONTRAINT_KIND_MISMATCH
```

Comment:
Excellent tests, thank you for the split!

One more request: could write a test with multiple type parameters to ensure that the formals and actuals list are checked in the right order? E.g.,

```
module M {
   resource R { }
   resource Test<T1: resource, T2: unrestricted> { f1: T1, f2: T2 }

    foo() {
        let x: Self.Test<Self.R, bool>;
        return;
    }
}
```